### PR TITLE
HRS 11

### PR DESCRIPTION
**Pull request by sherif98**
## Description:

Please include a summary of the changes, which issue is fixed, and include relevant motivation and context.

[Horse should make adjustment in stocks according to inventory level changes](https://rubikal.atlassian.net/browse/HRS-11)

## Type of change:
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing tests pass locally with my changes.
- [x] I have included a list of tasks/migrations that need to be run.

## Tasks/Migrations:

rails db:migrate

### CODE REVIEW

File: app/models/concerns/webhooks.rb [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    'products/delete',
+    'locations/create',
+    'locations/update',
+    'inventory_levels/connect',
+    'inventory_levels/update'
+  ].freeze
+
+  def self.webhook_path(store_id)
+    "#{ENV['ROOT_URL']}#{Rails.application
+														 .routes.url_helpers
```
Comment (hrdwdmrbl) [inline app/models/concerns/webhooks.rb]: nit: emptyn line

File: app/models/concerns/webhooks.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+				.webhooks_path(store_id: store_id)}"
+  end
+
+  def self.initialize_shopify_product(webhook_params)
+    shopify_product = ShopifyAPI::Product.new
+    shopify_product.attributes = webhook_params.except(:variants)
+    shopify_product.variants =
+      initialize_shopify_variants(webhook_params[:variants])
```
Comment (hrdwdmrbl) [inline app/models/concerns/webhooks.rb]: What's your line-length max? I usually go with 120

File: app/models/inventory_item.rb [hunk 3]
```diff
@@ -17,4 +17,15 @@ class InventoryItem < ApplicationRecord
   has_many :inventory_levels
   belongs_to :product_variant
   belongs_to :store
+
+  def self.sync!(shopify_variant, store_id)
+    inventory_item = where(
```
Comment (hrdwdmrbl) [inline app/models/inventory_item.rb]: find_by_id

File: app/models/inventory_level.rb [hunk 4]
```diff
@@ -25,5 +25,52 @@ class InventoryLevel < ApplicationRecord
   belongs_to :shared_inventory_level, optional: true
   belongs_to :store
   belongs_to :product_variant
-  belongs_to :shared_product_variant
+  belongs_to :shared_product_variant, optional: true
+
+  after_update :update_other_inventory_levels
+
+  def self.sync!(shopify_inventory_level, store_id)
```
Comment (hrdwdmrbl) [inline app/models/inventory_level.rb]: Can you comment what all the functions in this class do, it isn't quick to understand

File: db/migrate/20190407161725_rename_location_title.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class RenameLocationTitle < ActiveRecord::Migration[5.2]
+  def change
+		rename_column :locations, :title, :name
+
```
Comment (hrdwdmrbl) [inline db/migrate/20190407161725_rename_location_title.rb]: weird spacing

File: db/migrate/20190407174459_rename_inventory_level_quantity.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+class RenameInventoryLevelQuantity < ActiveRecord::Migration[5.2]
+  def change
+		rename_column :inventory_levels, :quantity, :available
+		rename_column :shared_inventory_levels, :quantity, :available
```
Comment (hrdwdmrbl) [inline db/migrate/20190407174459_rename_inventory_level_quantity.rb]: too much indent

File: app/models/concerns/webhooks.rb (hunk 2 above)
Comment (sherif98) [reply app/models/concerns/webhooks.rb]: We are following rubocop conventions which is 80.

Comment (hrdwdmrbl) [reply app/models/concerns/webhooks.rb]: Oh well, that's okay. The important thing is that you are using a linter :)

**Pull request merged**